### PR TITLE
Allow search without specifying anns_field

### PR DESCRIPTION
**Pull request by czs007**
issue #24475

### CODE REVIEW

File: pkg/util/typeutil/schema.go [hunk 1]
```diff
@@ -689,6 +689,16 @@ func MergeFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData) error
 	return nil
 }
 
+// GetVectorFieldSchema get vector field schema from collection schema.
+func GetVectorFieldSchema(schema *schemapb.CollectionSchema) (*schemapb.FieldSchema, error) {
+	for _, fieldSchema := range schema.Fields {
+		if fieldSchema.DataType == schemapb.DataType_BinaryVector || fieldSchema.DataType == schemapb.DataType_FloatVector {
```
Comment (xige-16) [inline pkg/util/typeutil/schema.go:695]: using typeutil.IsVectorType would be better

**Comment by xige-16** [comment]
What if milvus supports multiple vector fields in the future?

File: pkg/util/typeutil/schema.go (hunk 1 above)
Comment (czs007) [reply pkg/util/typeutil/schema.go:695]: ok

**Comment by czs007** [comment]
duplicated with https://github.com/milvus-io/milvus/pull/24499

**Pull request closed without merging**